frontend/src/components/user-details/user-profile/components/user-profile-form/user-profile-form.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  FC,
+  FormEvent,
+  UserDetailsResponseDto,
+  UserDetailsUpdateInfoRequestDto,
+} from 'common/types/types';
+import { Button, Input, Switcher } from 'components/common/common';
+import { getNameOf } from 'helpers/helpers';
+import { useAppForm } from 'hooks/hooks';
+import { userDetailsUpdateInfo as userDetailsUpdateInfoValidationSchema } from 'validation-schemas/validation-schemas';
+
+import { DEFAULT_UPDATE_USER_DETAILS_PAYLOAD } from './common';
+import styles from './styles.module.scss';
+
+type Props = {
+  userDetails: UserDetailsResponseDto | null;
+  onHandleUpdateProfile: (payload: UserDetailsUpdateInfoRequestDto) => void;
+};
+
+const UserProfileForm: FC<Props> = ({ userDetails, onHandleUpdateProfile }) => {
+  const { control, errors, setValue, getValues } =
+    useAppForm<UserDetailsUpdateInfoRequestDto>({
+      defaultValues: userDetails
+        ? userDetails
```

Comment:
Looks like `dateOfBirth` is not prepopulated from submitted value because its format returned by response (`MM-DD-YYYY T:Z`) is not acceptable by date input field

Reply:
@andanfe thank you for notice, it helped me)